cmk/base/plugins/agent_based/cisco_asa_failover.py

```diff
@@ -14,142 +15,147 @@
     Service,
+    Result,
+    State,
     SNMPTree,
     startswith,
-    State,
-)
-from cmk.base.plugins.agent_based.agent_based_api.v1.type_defs import (
-    CheckResult,
-    DiscoveryResult,
-    StringTable,
+    contains,
+    any_of,
 )
 
 
+def get_cisco_asa_state_name(st: str) -> str:
+    names = {
+        '1': 'other',
+        '2': 'up',
+        '3': 'down',
+        '4': 'error',
+        '5': 'overTemp',
+        '6': 'busy',
+        '7': 'noMedia',
+        '8': 'backup',
+        '9': 'active',
+        '10': 'standby',
+    }
+    return names.get(st, 'unknown %s' % st)
+
+
 @dataclass
 class Section:
     local_role: str
     local_status: str
     local_status_detail: str
     failover_link_status: str
+    failover_link_name: str
     remote_status: str
 
 
-def parse_cisco_asa_failover(string_table: StringTable) -> Optional[Section]:
+def parse_cisco_asa_failover(string_table: List[StringTable]) -> Optional[Section]:
     failover = {}
-    for role, status, detail in string_table:
-        if 'this device' in role and detail.lower() != 'failover off':
+    for role, status, detail in string_table[0]:
+        if 'this device' in role and not detail.lower() == 'failover off':
             failover['local_role'] = role.split(' ')[0].lower()
             failover['local_status'] = status
             failover['local_status_detail'] = detail
         elif 'failover' in role.lower():
             failover['failover_link_status'] = status
+            failover['failover_link_name'] = detail
         else:
             failover['remote_status'] = status
 
-    if len(failover) == 5:
+    try:
         return Section(
             local_role=failover['local_role'],
             local_status=failover['local_status'],
             local_status_detail=failover['local_status_detail'],
             failover_link_status=failover['failover_link_status'],
+            failover_link_name=failover['failover_link_name'],
             remote_status=failover['remote_status'],
         )
-    return None
+    except KeyError:
+        return None
 
 
 def discovery_cisco_asa_failover(section: Section) -> DiscoveryResult:
     yield Service()
 
 
-_STATE_NAMES = {
-    '1': 'other',
-    '2': 'up',
-    '3': 'down',
-    '4': 'error',
-    '5': 'overTemp',
-    '6': 'busy',
-    '7': 'noMedia',
-    '8': 'backup',
-    '9': 'active',
-    '10': 'standby',
-}
-
-
-def _get_cisco_asa_state_name(st: str) -> str:
-    return _STATE_NAMES.get(
-        st,
-        f'unknown {st}',
-    )
-
-
 def check_cisco_asa_failover(params: (Mapping[str, Any]), section: Section) -> CheckResult:
-    yield Result(
-        state=State.OK,
-        summary=f'Device ({section.local_role}) is the {section.local_status_detail}',
-    )
+    yield Result(state=State.OK,
+                 summary='Device (%s) is the %s' % (section.local_role, section.local_status_detail))
 
-    if params[section.local_role] != _get_cisco_asa_state_name(section.local_status):
```

Comment:
Why did you modify this (the formatting looked much better before and f-strings are also better)?